core/src/test/scala/org/apache/spark/deploy/security/HadoopDelegationTokenManagerSuite.scala

```diff
@@ -111,6 +113,17 @@ class HadoopDelegationTokenManagerSuite extends SparkFunSuite with Matchers {
     creds.getAllTokens.size should be (0)
   }
 
+  test("Obtain tokens For Kafka") {
+    val hadoopConf = new Configuration()
+    sparkConf.set(KAFKA_DELEGATION_TOKEN_ENABLED, true)
+
+    val kafkaTokenProvider = new KafkaDelegationTokenProvider()
+    val creds = new Credentials()
+    kafkaTokenProvider.obtainDelegationTokens(hadoopConf, sparkConf, creds)
+
+    creds.getAllTokens.size should be (0)
```

Comment:
I have seen this is follows the `Obtain tokens For HBase` but the using `creds.getAllTokens.size should be (0)` for both the positive and negative case (if I understand correctly) is quite strange. Can you somehow mock a relevant method (i.e. in TokenUtil) to get back something?

Reply:
I was thinking about this and don't see clean option to do it:
* Adding kafka dependency to `HadoopDelegationTokenManagerSuite` is bad idea.
* `KafkaMicroBatchSourceSuite` can't reach neither `HadoopDelegationTokenManager` nor `KafkaDelegationTokenProvider`.
* KDC is not available in general.

Reply:
I see. Unfortunately this way this test has not much value as in obtainDelegationTokens an exception is thrown when the loadclass is called with "org.apache.spark.sql.kafka010.TokenUtil". But as this is quite the same for HBase you can keep it.

Reply:
Agree with Attila that this is not helping much. If you can have better tests on the kafka connector side, and exercise the `KafkaDelegationTokenProvider` code from there, it would be better.

Reply:
Increased test coverage.